src/quota.rs

```diff
@@ -0,0 +1,42 @@
+use crate::grpc_sys::{self, grpc_resource_quota};
+use std::ptr;
+
+/// ResourceQuota represents a bound on memory and thread usage by the gRPC,
+/// A ResourceQuota need to be ensure lifetime when the attached Channel is
+/// working or the program is aborted.
+pub struct ResourceQuota {
+    raw: *mut grpc_resource_quota,
+}
+
+impl ResourceQuota {
+    /// Create a control block for resource quota
+    pub fn new(name: Option<&str>) -> ResourceQuota {
+        match name {
+            Some(name_str) => ResourceQuota {
+                raw: unsafe { grpc_sys::grpc_resource_quota_create(name_str.as_ptr() as _) },
```

Comment:
I don't think this is safe - the name may not live long enough. I think it is best to only take `&'static str`

Reply:
And I think the string should be null-terminated, so either take a C str or make it unsafe and add a comment

Reply:
The method will clone the string. But null-terminated is a good point! 👍